parsil/src/executor.rs

```diff
@@ -107,6 +169,32 @@ impl AstPass for TranslatedQuery {
     }
 }
 
+struct PlaceholderInterpolator<'a, C: ContextProvider> {
+    settings: &'a ParsilSettings<C>,
+    placeholders: &'a Placeholders,
+}
+impl<'a, C: ContextProvider> AstPass for PlaceholderInterpolator<'a, C> {
+    fn post_expr(&mut self, expr: &mut Expr) -> Result<()> {
+        if let Some(replacement) = if let Expr::Value(Value::Placeholder(name)) = expr {
+            let value = self
+                .placeholders
+                .get(&self.settings.placeholders.resolve_placeholder(name)?)?;
+            Some(Expr::Cast {
+                kind: CastKind::DoubleColon,
+                expr: Box::new(Expr::Value(Value::SingleQuotedString(format!("{value}")))),
```

Comment:
I have hard time understanding if this is safe or not lol wdyt ?

Reply:
It is; this is the PgSQL way of expressing `'123456789123456789'::NUMERIC`.